### DISCUSSION TITLE
Support for using cognito in the alb listner

### DISCUSSION THREAD

**Discussion by sewmiuraj** [gruntwork-io/knowledge-base#758] [How do I] [answered]
Hello,

I wish to add authentication using cognito in my public loadbalancer. However, when I look at https://github.com/gruntwork-io/terraform-aws-load-balancer/blob/main/modules/lb-listener-rules/main.tf, I don't see an option to be able to specify type             = "authenticate-cognito". 

Have I missed anything, or is there a way to be able to configure this?

---

<ins datetime="2023-08-03T14:03:26Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110382">Tracked in ticket #110382</a></p>
</ins>

**Comment by gcagle3** [comment]
Hi @sewmiuraj! At present, this module only supports `type = "authenticate-oidc"`. If you could explain your use case a bit (clarify how you're wanting to leverage this and what problem you're trying to solve), then we would be happy to open an issue to add this functionality to the module. Thanks in advance!

**Reply by sewmiuraj** [reply]
Hello @gcagle3 ,

I have a setup where I have integrated Azure AD with AWS Congito. This lets me configure authentication to the application at an alb level.  I already have my implementation done manually through the AWS console. While trying to update the configuration in the terragrunt files, I realized that type = "authenticate-cognito" is not supported.

**Reply by gcagle3** [reply]
Hi @sewmiuraj, we have opened a [feature request issue here](https://github.com/gruntwork-io/terraform-aws-load-balancer/issues/188) to add Cognito authentication as an option. If you have any additional or specific criteria you would like to add, please feel free to add it to the issue as a comment. We should be able to start work on this soon.

**Reply by gcagle3** [reply]
Hi @sewmiuraj! There is a feature branch [opened here](https://github.com/gruntwork-io/terraform-aws-load-balancer/tree/feature/188-lb-listener-rules-support-for-cognito) to add Cognito authentication support to the `lb-listener-rules` module. If you have time, could you test this out and let us know if this works as expected?

**Comment by gcagle3** [comment, marked as answer]
Hi @sewmiuraj! I'm happy to report that Cognito support is officially available in terraform-aws-load-balancer [v0.29.12](https://github.com/gruntwork-io/terraform-aws-load-balancer/releases/tag/v0.29.12) now. I'm going to go ahead and close this discussion, but please feel free to re-open it (or a new discussion) if you have any questions.
